adaptive_alerting_detector_build/datasources/__init__.py

```diff
@@ -0,0 +1,55 @@
+import json
+import requests
+import pandas as pd
+import numpy as np
+import datetime
+from adaptive_alerting_detector_build.exceptions import AdaptiveAlertingDetectorBuildError
+
+
+class _datasource:
+    def __init__(self, **kwargs):
+        pass
+
+    def query(self):
+        raise NotImplementedError
+
+
+class DatasourceQueryException(AdaptiveAlertingDetectorBuildError):
+    """Raised when query fails."""
+
+
+class graphite(_datasource):
```

Comment:
can we move this as separate class file?
datasource
   |-DatasourceFactoryProvider.py 
    |-graphite.py
in that way, it will be more modular to easy to add new data source later on rather than using Datasource logic in ___int__.py file

Reply:
we can do it later if needed...or now.  I don't think the file will get too large even if we keep them together.

Reply:
I started it that way, but felt is was easier to manage like this.